docker/install_node.sh

```diff
@@ -54,6 +61,10 @@ if [ "$node_file_name" = "node-v16.13.0" ]; then
     make
 fi
 
-cd ../oppia_tools && find . -maxdepth 1 -type d -name 'node*' -exec mv {} node-16.13.0 \;
+# Rename node directory to node-16.13.0.
```

Comment:
Shouldn't we do this only if the node file has just been downloaded? I.e. after rm node-download above perhaps.

Reply:
I don't think it is required as all lines after line 23 (# Install node.js if not installed) are exectuted only if node isn't installed and are skipped if node is already installed.
Also if lines are used below rm node-download, it will only execute for linux installations and not for windows. If we want, we can move this line after if command ends in line 56 (fi), but not inside if block.

(@seanlip is it good now.)

Reply:
Thanks -- yup, you put the comment in the right place this time.

Yes, I would suggest moving it after `fi` as suggested. Then you wouldn't need the "if" on line 58, correct?

Though TBH I am a bit confused about when lines 58-62 in the current code are supposed to be executed. Is it only meant to be done for non-Windows/x64 systems?

Reply:
I already commented here before but also marked as resolved, so was confused.
Yeah! You are right code in line 58-62 is only meant for non-windows/x64 systems and thus CAN BE moved moved inside if block, and my changes can we added below line 57 (outside if block).

Reply:
@jayam04 I'm not sure I follow. Shouldn't we run  .configure and make for all systems, not just non-Windows/x64 ones? If not, why not?

Reply:
@seanlip we don't require to do it for windows, Linux or Darwin as their node JS code comes pre-compiled. But when using source code (which we are using only for nonWindows/x64) we need to compile it based on machine config and thus `./configure` and `make` are required. You can find reference [here](https://github.com/nodejs/node/blob/main/BUILDING.md#building-nodejs-1)

Reply:
@jayam04 I think we can remove it (line 52-61) since we plan to support: Windows, Linux, and Mac OS (Darwin) since docker supports these platforms. Do you have any system that may not fall into these categories?

Reply:
@gp201 Nah! I can't think of any other system currently. There is also a new problem - nowadays uname in windows results in Msys-someVersion thus script won't work. I checked it in windows 11. Should I open new Issue for it as system info in windows is quite complicated and new issue would serve better.

cc @seanlip for windows related issue.

Reply:
@jayam04 Thanks for identifying that. I would actually suggest fixing it once and for all in this PR. Can you try to support the most common unames? (It's fine if you don't get all of them, we'll fix those on the fly as needed.)

For @gp201's point I would be fine with either keeping that block of code and leaving a comment to explain that we only do this for specific OS's (say which those are) and pointing to the URL reference, or alternatively deleting that block altogether and throwing an error if the condition on line 40 is not satisfied. (But what is the condition on [line 40](https://github.com/oppia/oppia/pull/19373/files#diff-edbed02d7e52cfd3985c519ff114c72a7303b127cfeea826a3b7cb5e0fd33991R40) checking exactly?)

Reply:
@gp201 I think we should keep code, cause it's for non-x64 linux/darwin system.
@seanlip line 40 check if current system is x64 or not.
Also can you help me which e2e checks failing. Is it because of code being wrong (which I don't think so, checked it) or due to huge changes in code or any other reason.

Reply:
@jayam04 We discourage force-pushing. Why are you doing this and why does regular pushing not work for you?

The e2e test in question seems like it might be flaky. I restarted it, but it's not related to your PR.

I'm fine with keeping the code but please add a comment as previously requested.

Reply:
@seanlip is all done now??

Reply:
@jayam04 Please answer my first question.

Reply:
Ah! I see. Do you mean me rebasing branch with develop?

Reply:
Yes. Please follow the PR guide, this is mentioned in the fourth bullet point [here](https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve). Thanks.

Reply:
> @gp201 I think we should keep code, cause it's for non-x64 linux/darwin system. @seanlip line 40 checks if current system is x64 or not. Also can you help me with e2e checks failing. Is it because of code being wrong (which I don't think so, checked it) or due to huge changes in code or any other reason.

@jayam04 Can you add a comment explaining the same? Saying that we need it for certain systems (sys1, sys2 etc...).

Reply:
@gp201 I think it's already done, can you check line 59 in current code. Isn't it enough to say non-x64 linux/darwin systems.